protos/flyteidl/core/security.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package flyteidl.core;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core";
+
+// Secret encapsulates information about the secret a task needs to proceed. An environment variable
+// FLYTE_SECRETS_ENV_PREFIX will be passed to indicate the prefix of the environment variables that will be present if
+// secrets are passed through environment variables.
+// FLYTE_SECRETS_DEFAULT_DIR will be passed to indicate the prefix of the path where secrets will be mounted if secrets
+// are passed through file mounts.
+message Secret {
+    enum Type {
+        // Symmetric indicates a shared secret (e.g. password).
+        Symmetric = 0;
+
+        // Asymmetric indicates a public-private key where the reference is to the private portion of the key.
+        Asymmetric = 1;
+    }
+
+    // The name of the secret to mount. This has to match an existing secret in the system. It's up to the implementation
+    // of the secret management system to require case sensitivity.
+    string name = 1;
+
+    // Type indicates the type of key requested. Asymmetric keys can not be passed through environment variables. If the
```

Comment:
this is not true for Confidant for example - right? So instead do we want to say that envvar injection possible or no?

Reply:
done